wal/decoder.go

```diff
@@ -98,6 +113,44 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	return nil
 }
 
+// isTornEntry determines whether the last entry of the WAL was partially written
+// and corrupted because of a torn write.
+func (d *decoder) isTornEntry(data []byte) bool {
+	if len(d.brs) != 1 {
+		return false
+	}
+
+	fileOff := d.lastValidOff + 8
+	curOff := 0
+	chunks := [][]byte{}
+	// split data on sector boundaries
+	for curOff < len(data) {
+		chunkLen := int(512 - (fileOff % 512))
```

Comment:
we assume the sector size is 512 here? 4096 should be ok though. probably add a minSectorSize const